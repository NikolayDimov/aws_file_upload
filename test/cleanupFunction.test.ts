import { Stack } from "aws-cdk-lib";
import { Table, AttributeType } from "aws-cdk-lib/aws-dynamodb";
import { Topic } from "aws-cdk-lib/aws-sns";
import "jest-cdk-snapshot";

describe("Infrastructure Snapshot Test", () => {
    it("should match the CloudFormation snapshot", () => {
        const stack = new Stack();

        new Table(stack, "ErrorTable", {
            partitionKey: { name: "id", type: AttributeType.STRING },
        });

        new Topic(stack, "ErrorTopic");

        expect(stack).toMatchCdkSnapshot();
    });
});
