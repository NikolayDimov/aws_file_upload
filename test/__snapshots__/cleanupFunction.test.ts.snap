// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Infrastructure Snapshot Test should match the CloudFormation snapshot 1`] = `
{
  "Resources": {
    "ErrorTableA27F7F12": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "AttributeDefinitions": [
          {
            "AttributeName": "id",
            "AttributeType": "S",
          },
        ],
        "KeySchema": [
          {
            "AttributeName": "id",
            "KeyType": "HASH",
          },
        ],
        "ProvisionedThroughput": {
          "ReadCapacityUnits": 5,
          "WriteCapacityUnits": 5,
        },
      },
      "Type": "AWS::DynamoDB::Table",
      "UpdateReplacePolicy": "Retain",
    },
    "ErrorTopicA0904A23": {
      "Type": "AWS::SNS::Topic",
    },
  },
}
`;
